### PR TITLE
Update Kotlin to 1.4.10 version.

### PR DESCRIPTION
**Pull request by Tapchicoma**
Drop support for deprecated Kotlin plugins.

First of planned breaking changes for next `10.0.0` release.

### CODE REVIEW

File: plugin/build.gradle.kts [hunk 1]
```diff
@@ -21,6 +22,12 @@ repositories {
     jcenter()
 }
 
+tasks.withType<KotlinCompile>() {
+    kotlinOptions {
+        apiVersion = "1.3"
```
Comment (JLLeitschuh) [inline plugin/build.gradle.kts:27]: Hrm... Gradle is still pretty heavily reliant upon version 1.2. Correct? If we start using 1.3 features, we'll break backwards compatibility with older versions of Gradle, won't we?

Comment (Tapchicoma) [reply plugin/build.gradle.kts:27]: apiVersion `1.3` was default target for the code compiled with Kotlin version `1.3.*`. Also Gradle itself is using Kotlin `1.3.*` for kotlin scripts, so I assume `1.3` here is correct.

Also samples works correctly.

Comment (JLLeitschuh) [reply plugin/build.gradle.kts:27]: Oh, I was thinking about Kotlin 1.4
Nevermind. Thanks!

**Review by JLLeitschuh — APPROVED** [review]
LGTM!

**Pull request merged**